Documentation/dev-internal/modules.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Modules
+---
+
+The etcd project (since version 3.5) is organized into multiple 
+[golang modules](https://golang.org/ref/mod) hosted  in a [single repository](https://golang.org/ref/mod#vcs-dir).
+
+![modules graph](img/modules.svg)
+
+There are following modules: 
+  
+  - **go.etcd.io/etcd/api/v3** - contains API definitions
+  (like protos & proto-generated libraries) that defines communication protocol
+  between etcd clients and server.
+  
+  - **go.etcd.io/etcd/pkg/v3** - collection of utility packages used by etcd
+  without being specific to etcd itself. A package belongs here
+  only if it could possibly be moved out into its own repository in the future.
+  Please avoid adding here code that has a lot of dependencies on its own, as
+  they automatically becoming dependencies of the client library
+  (that we want to keep lightweight).
+
+  - **go.etcd.io/etcd/client/v3** - client library used to contact etcd over 
+  the network (grpc). Recommended for all new usage of etcd. 
+  
+  - **go.etcd.io/etcd/client/v2** - legacy client library used to contact etcd 
+  over HTTP protocol. Deprecated. All new usage should depend on /v3 library.
+  
+  - **go.etcd.io/etcd/raft/v3** - implementation of distributed consensus 
+  protocol. Should have no etcd specific code.
+  
+  - **go.etcd.io/etcd/server/v3** - etcd implementation.
+  The code in this package is etcd internal and should not be consumed 
+  by external projects. The package layout and API can change within the minor versions.
+  
+  - **go.etcd.io/etcd/etcdctl/v3** - a command line tool to access and manage etcd. 
+    We reserve the right to split a tool to manage etcd-files (snapshots, dumps) to a separate 
```

Comment:
The project reserves the right to make changes to any tool or package in general so suggest to either re-write this sentence or simply remove it because in the `Future` section you have already mentioned it. Thanks!